manifests/virt-apiserver.yaml.in

```diff
@@ -0,0 +1,195 @@
+  name: "kubevirt.io:virt-apiserver"
+  apiGroup: rbac.authorization.k8s.io
+---
+# apiserver gets the auth-delegator role to delegate auth decisions to
+# the core apiserver
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: "kubevirt.io:apiserver-auth-delegator"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+- apiGroup: ""
+  kind: ServiceAccount
+  name: virt-apiserver
+  namespace: default
+---
+# apiserver gets the ability to read authentication. This allows it to
+# read the specific configmap that has the requestheader-* entries to
+# enable api aggregation
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: "kubevirt.io:apiserver-authentication-reader"
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: extension-apiserver-authentication-reader
+subjects:
+- apiGroup: ""
+  kind: ServiceAccount
+  name: virt-apiserver
+  namespace: default
+---
+apiVersion: apiregistration.k8s.io/v1beta1
+kind: APIService
+metadata:
+  name: v1alpha1.kubevirt.io
+  insecureSkipTLSVerify: true
+  #service catalog skipped these values
+  labels:
+    api: virt-apiserver
+    apiserver: "true"
+spec:
+  caBundle: "{{ apiserver_tls_crt }}"
+  group: kubevirt.io
+  version: v1alpha1
+  # service catalog left out priority for apiregistration.k8s.io/v1beta1
+  groupPriorityMinimum: 2000
+  versionPriority: 20
+  priority: 200
+  service:
+    name: virt-apiserver-service
+    namespace: default
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: virt-apiserver-service
+spec:
+  ports:
+    - port: 443
+      protocol: TCP
+      targetPort: virt-apiserver
+  externalIPs :
+    - "{{ master_ip }}"
+  selector:
+    app: virt-apiserver
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: virt-apiserver
+spec:
+  template:
+    metadata:
+      labels:
+        app: virt-apiserver
+    spec:
+      containers:
+      - name: virt-apiserver
+        image: {{ docker_prefix }}/virt-apiserver:{{ docker_tag }}
+        imagePullPolicy: IfNotPresent
+        command:
+            - "/virt-apiserver"
+            - "--bind-address"
+            - "0.0.0.0"
+            - "--secure-port"
+            - "8183"
+            - "--tls-cert-file=/var/run/virt-apiserver-cert/virt-apiserver.crt"
+            - "--tls-private-key-file=/var/run/virt-apiserver-cert/virt-apiserver.key"
+            - "--requestheader-client-ca-file=/var/run/virt-apiserver-cert/requestheader-ca.crt"
+            - "--etcd-servers=http://127.0.0.1:2379"
+            - "--authentication-skip-lookup"
+            - "-v"
+            - "9"
+        ports:
+          - containerPort: 8183
+            name: "virt-apiserver"
+            protocol: "TCP"
+        volumeMounts:
+          - mountPath: /var/run/virt-apiserver-cert
+            name: virt-apiserver-cert
+      - name: etcd
```

Comment:
What is the reason for having a separate instance of `etcd` can't the API server use the same instance than the rest of the cluster? If the administrator of the cluster configures it for high availability and replication of the master, then this separate instance of `etcd` won't benefit from that, if I understand correctly.